src/server/game/Spells/SpellEffects.cpp

```diff
@@ -2759,15 +2759,20 @@ void Spell::EffectTeleUnitsFaceCaster(SpellEffIndex  /*effIndex*/)
     if (!unitTarget || unitTarget->IsInFlight())
         return;
 
-    if (!m_targets.HasDst())
+    if (unitTarget->IsInFlight())
     {
-        sLog->outError("Spell::EffectTeleUnitsFaceCaster - does not have destination for spell ID %u\n", m_spellInfo->Id);
         return;
     }
 
-    float x, y, z;
-    destTarget->GetPosition(x, y, z);
-    unitTarget->NearTeleportTo(x, y, z, unitTarget->GetAngle(m_caster));
+    if (m_targets.HasDst())
+    {
+        unitTarget->NearTeleportTo(destTarget->GetPositionX(), destTarget->GetPositionY(), destTarget->GetPositionZ(), destTarget->GetAbsoluteAngle(m_caster), unitTarget == m_caster);
+    }
+    else
+    {
+        sLog->outError("Spell::EffectTeleUnitsFaceCaster - does not have destination for spell ID %u\n", m_spellInfo->Id);
+        return;
```

Comment:
It should be able to be deleted here？

Reply:
I thought so too, but didn't know if !m_targets.HasDst() can still happen

Reply:
It should be OK without it, because there is no code below

Reply:
agree, I think we can remove that `return`

Reply:
okidoki, thanks ^^

Do we need the outError?

Reply:
I don't think so

Reply:
Thanks for an answer. Now I can finish it :)